v-next/hardhat/test/src/internal/builtin-plugins/run/runScriptWIthHardhat.ts

```diff
@@ -0,0 +1,110 @@
+import type { HardhatRuntimeEnvironment } from "@nomicfoundation/hardhat-core/types/hre";
+
+import assert from "node:assert/strict";
+import { before, describe, it } from "node:test";
+
+import { createHardhatRuntimeEnvironment } from "@nomicfoundation/hardhat-core";
+import { HardhatError } from "@nomicfoundation/hardhat-errors";
+
+import { runScriptWithHardhat } from "../../../../../src/internal/builtin-plugins/run/runScriptWithHardhat.js";
+import { useFixtureProject } from "../../../../helpers/project.js";
+
+describe("runScriptWithHardhat", function () {
+  let hre: HardhatRuntimeEnvironment;
+
+  before(async function () {
+    hre = await createHardhatRuntimeEnvironment({});
+  });
+
+  it("should throw if script is not a string", async function () {
+    await assert.rejects(async () => {
+      await runScriptWithHardhat({ script: 123, noCompile: false }, hre);
+    }, /An internal invariant was violated: Expected script to be a string/);
```

Comment:
very minor, but for future reference `assert.rejects` can take a promise, so this can be simplified a bit:

```suggestion
    await assert.rejects(
      runScriptWithHardhat({ script: 123, noCompile: false }, hre),
      /An internal invariant was violated: Expected script to be a string/);
```

Reply:
I blame copilot :)